### PR TITLE
FIO test using ripsaw tool

### PR DESCRIPTION
**Pull request by keesturam**
This PR covers the FIO storage performance test based on the ripsaw benchmark operator.

### CODE REVIEW

File: ocs_ci/templates/workloads/fio/customresource.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+apiVersion: ripsaw.cloudbulldozer.io/v1alpha1
```
Comment (RazTamir) [inline ocs_ci/templates/workloads/fio/customresource.yaml:1]: Please change the file name to more meaningful

**Comment by keesturam** [comment]
> Looks great Karthick!
> 
> The parametrize execution will iterate on the custom resource file?
Different block sizes (1024ki, 64ki, 4ki) defined in the CR file are run as a single test. We will parameterize on the interface type (rbd, cephfs)
> Are we going to expand it?
Not sure, which part you are asking for. But, so far, the plan is to cover the following
1) Different workload types (sequential read, sequential write, random read, random write, random read-write)
2) Different % of storage capacity used
3) Compare against a benchmark and fail if there is a regression
4) update a google sheet with a summary (% of regression or improvement against the benchmark)

**Comment by vasukulkarni** [comment]
👍

**Comment by RazTamir** [comment]
> > Looks great Karthick!
> > The parametrize execution will iterate on the custom resource file?
> 
> Different block sizes (1024ki, 64ki, 4ki) defined in the CR file are run as a single test. We will parameterize on the interface type (rbd, cephfs)
> 
> > Are we going to expand it?
> > Not sure, which part you are asking for. But, so far, the plan is to cover the following
> 
> 1. Different workload types (sequential read, sequential write, random read, random write, random read-write)
> 2. Different % of storage capacity used
> 3. Compare against a benchmark and fail if there is a regression
> 4. update a google sheet with a summary (% of regression or improvement against the benchmark)

+100!

File: tests/e2e/performance/test_fio_benchmark.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    # Create RipSaw Operator
+    ripsaw = RipSaw()
+
+    def teardown():
+        ripsaw.cleanup()
+    request.addfinalizer(teardown)
+    return ripsaw
+
+
```
Comment (ebenahar) [inline tests/e2e/performance/test_fio_benchmark.py:30]: polarion id?

File: ocs_ci/templates/workloads/fio/customresource.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        - 4KiB
+      numjobs:
+        - 1
+      iodepth: 4
+      read_runtime: 60
+      read_ramp_time: 5
+      filesize: 2GiB
+      log_sample_rate: 1000
+      storageclass: ocs-storagecluster-ceph-rbd
+      storagesize: 5Gi
```
Comment (Avilir) [inline ocs_ci/templates/workloads/fio/customresource.yaml:34]: I think we should work on larger PV size, and with larger filesize - on line 31

**Comment by jtaleric** [comment]
We should put this PR on hold until https://github.com/cloud-bulldozer/ripsaw/pull/321 is merged since 321 changes the CR a bit.

**Comment by keesturam** [comment]
> We should put this PR on hold until [cloud-bulldozer/ripsaw#321](https://github.com/cloud-bulldozer/ripsaw/pull/321) is merged since 321 changes the CR a bit.

This PR is now merged.

File: tests/e2e/performance/test_fio_benchmark.py (hunk 2 above)
Comment (keesturam) [reply tests/e2e/performance/test_fio_benchmark.py:32]: Added now.

File: ocs_ci/templates/workloads/fio/customresource.yaml (hunk 1 above)
Comment (keesturam) [reply ocs_ci/templates/workloads/fio/customresource.yaml]: changed to `fio_cr_template.yaml`

File: ocs_ci/templates/workloads/fio/cr_template.yaml [hunk 4]
```diff
@@ -0,0 +1,82 @@
+apiVersion: ripsaw.cloudbulldozer.io/v1alpha1
```
Comment (ebenahar) [inline ocs_ci/templates/workloads/fio/cr_template.yaml:1]: `cr_template.yaml` is too generic name IMO

File: tests/e2e/performance/test_fio_benchmark.py [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+def analyze_regression(io_pattern, es_username):
+    """
```
Comment (ebenahar) [inline tests/e2e/performance/test_fio_benchmark.py:33]: please add a function description above `Args`

File: tests/e2e/performance/test_fio_benchmark.py [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    Args:
+        io_pattern (str): 'sequential' or 'random' workload
+        es_username (str): ocs_build used in the CR object
```
Comment (ebenahar) [inline tests/e2e/performance/test_fio_benchmark.py:36]: please add an empty line below

**Comment by keesturam** [comment]
Here are the results from elasticsearch of a sample run - 

cephFS - Random read and Ramdom write - http://10.0.78.167:3000/d/TAL8V_PZz/new-aws-fio-summary-with-rook-ceph?orgId=1&from=1590139667360&to=1590172905830&var-user=bm_test_random&var-clustername=AWS4.4.04.4.0-420.ci&var-UUID=acf77640-b787-530c-b3eb-c6f5509f9dae&var-Operation=All&var-io_size=All&var-interval=$__auto_interval_interval

RBD - random read and random write - http://10.0.78.167:3000/d/TAL8V_PZz/new-aws-fio-summary-with-rook-ceph?orgId=1&from=1590119397761&to=1590145728236&var-user=bm_test_random&var-clustername=AWS4.4.04.4.0-420.ci&var-UUID=1ca0d29e-c9a9-5cec-89fd-63df1a5c60da&var-Operation=All&var-io_size=All&var-interval=$__auto_interval_interval

File: ocs_ci/ocs/ocp.py [hunk 7]
```diff
@@ -928,6 +928,20 @@ def get_build():
     return ocp_cluster.get()['items'][0]['status']['desired']['version']
 
 
+def get_platform():
```
Comment (Avilir) [inline ocs_ci/ocs/ocp.py:931]: You can use from node.py the get_provider() method.

File: ocs_ci/ocs/ripsaw.py [hunk 8]
```diff
@@ -75,14 +76,14 @@ def _clone_ripsaw(self):
...
             raise cf
 
-    def apply_crd(self, crd):
+    def apply_crd(self, crd, drop_cache):
         """
         Apply the CRD
 
         Args:
             crd (str): Name of file to apply
+            drop_cache (bool): Enable if OSD cache needs to be dropped
```
Comment (Avilir) [inline ocs_ci/ocs/ripsaw.py:85]: where you are using this?
i do not see and deployment of this pod (drop_cache)

File: tests/e2e/performance/test_fio_benchmark.py (hunk 5 above)
Comment (keesturam) [reply tests/e2e/performance/test_fio_benchmark.py:35]: This is done now.

File: tests/e2e/performance/test_fio_benchmark.py (hunk 6 above)
Comment (keesturam) [reply tests/e2e/performance/test_fio_benchmark.py:41]: Done

File: ocs_ci/ocs/ripsaw.py (hunk 8 above)
Comment (keesturam) [reply ocs_ci/ocs/ripsaw.py]: I've removed this.

File: ocs_ci/templates/workloads/fio/cr_template.yaml (hunk 4 above)
Comment (keesturam) [reply ocs_ci/templates/workloads/fio/cr_template.yaml]: This is now changed to `benchmark_fio.yaml`

File: tests/e2e/performance/test_fio_benchmark.py [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+        # Todo: Remove the below skip for random workload once
+        #  https://github.com/cloud-bulldozer/snafu/issues/180 is fixed
+        if io_pattern == 'sequential':
+            std_dev = 'std-dev-' + object_size
+            variance = test_data[std_dev]
+            log.warning(f'variance - {variance} is greater than 5%')
+        # Todo: Fail test if 5% deviation from benchmark value
+
+
+@pytest.mark.skipif(
```
Comment (vasukulkarni) [inline tests/e2e/performance/test_fio_benchmark.py:72]: @keesturam can you use this mark instead https://github.com/red-hat-storage/ocs-ci/pull/2119 , I will update the final comment I have there. The travis is failing due to get_plaftorm call

File: ocs_ci/ocs/ocp.py (hunk 7 above)
Comment (keesturam) [reply ocs_ci/ocs/ocp.py]: I've removed this function now.

**Comment by keesturam** [comment]
> PR validation on existing cluster [Job UNSTABLE](https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-trigger-test-pr/1905//testReport) (some or all tests failed).

one of the four tests has failed during the teardown. some PVCs were not deleted sooner during the teardown. I think this is mostly because of a slow test environment used for the test. Ideally, this should not be seen.

**Pull request merged**